pkg/core/blockchain.go

```diff
@@ -470,14 +471,13 @@ func (bc *Blockchain) GetTransaction(hash util.Uint256) (*transaction.Transactio
 
 	var height uint32
 	r.ReadLE(&height)
+
+	tx := &transaction.Transaction{}
+	tx.DecodeBinary(r)
 	if r.Err != nil {
```

Comment:
shouldn't we move error checking a bit higher?(just after ReadLE)

Reply:
since tx.DecodeBinary will fail anyway

Reply:
That's the whole point of this patchset --- to only check for errors once in the end of the whole deserialization process. It does mean that there could be some excessive work done, but I think we gain way more than we lose.

Reply:
true. I like the way it does. But afraid about common view: how usual default developer look at this function and used. That's why I wrote in general comment proposal about immutable stuff which we can discuss later. This one looks good as for me and I'm totally for to write less err!=nil =).

Reply:
I guess usual developer should refer to `io` package documentation if he doesn't understand the way it works, and there is something written about Bin* and Serializable there. So, yes, it's a bit unusual, but even then it's not completely alien, see, for example:
https://blog.golang.org/errors-are-values
This essentially is just an extension of the approach outlined in the article.